pancancer_utilities/scripts/classify_cancer_type.py

```diff
@@ -128,32 +144,54 @@
     hyper_filter=5
 )
 
+use_samples, rnaseq_df, y_df, gene_features = align_matrices(
+    x_file_or_df=rnaseq_df,
+    y=y_df,
+    add_cancertype_covariate=args.use_pancancer,
+    add_mutation_covariate=True
+)
+
 # shuffle mutation status labels if necessary
 if args.shuffle_labels:
     y_df.status = np.random.permutation(y_df.status.values)
 
+############################################
+### 4. Split data and fit/evaluate model ###
+############################################
+
 for fold_no in range(args.num_folds):
 
-    X_train_raw_df, X_test_raw_df = du.split_by_cancer_type(
-       rnaseq_df, sample_info_df, args.holdout_cancer_type,
-       num_folds=args.num_folds, fold_no=fold_no,
-       use_pancancer=args.use_pancancer, seed=args.seed)
+    logging.debug('Splitting data and preprocessing features...')
 
+    # split data into train and test sets
     try:
-        train_samples, X_train_df, y_train_df = align_matrices(
-            x_file_or_df=X_train_raw_df, y=y_df, add_cancertype_covariate=False
-        )
-        test_samples, X_test_df, y_test_df = align_matrices(
-            x_file_or_df=X_test_raw_df, y=y_df, add_cancertype_covariate=False
-        )
+        X_train_raw_df, X_test_raw_df = du.split_by_cancer_type(
+           rnaseq_df, sample_info_df, args.holdout_cancer_type,
+           num_folds=args.num_folds, fold_no=fold_no,
+           use_pancancer=args.use_pancancer, seed=args.seed)
     except ValueError:
         exit('No test samples found for cancer type: {}, gene: {}\n'.format(
                args.holdout_cancer_type, args.gene))
 
+    y_train_df = y_df.reindex(X_train_raw_df.index)
+    y_test_df = y_df.reindex(X_test_raw_df.index)
+
+    # data processing/feature selection, needs to happen for train and
+    # test sets independently
```

Comment:
Yes, nice thread you retweeted the other day https://twitter.com/jjc2718/status/1291182838612660224